sycl/source/detail/global_handler.cpp

```diff
@@ -228,20 +228,10 @@ ThreadPool &GlobalHandler::getHostTaskThreadPool() {
 
 void GlobalHandler::releaseDefaultContexts() {
   // Release shared-pointers to SYCL objects.
-#ifndef _WIN32
+  // Note that on Windows the destruction of the default context
+  // races with the detaching of the DLL object that calls piTearDown.
+
   MPlatformToDefaultContextCache.Inst.reset(nullptr);
-#else
-  // Windows does not maintain dependencies between dynamically loaded libraries
-  // and can unload SYCL runtime dependencies before sycl.dll's DllMain has
-  // finished. To avoid calls to nowhere, intentionally leak platform to device
-  // cache. This will prevent destructors from being called, thus no PI cleanup
-  // routines will be called in the end.
-  // Update: the pi_win_proxy_loader addresses this for SYCL's own dependencies,
-  // but the GPU device dlls seem to manually load yet another DLL which may
-  // have been released when this function is called. So we still release() and
-  // leak until that is addressed. context destructs fine on CPU device.
-  MPlatformToDefaultContextCache.Inst.release();
-#endif
```

Comment:
Do I get it right that you have to modify all the tests because this workaround is removed? Do we have to remove it to make default context as default.

Reply:
This workaround (or not) has no bearing on the leak check itself. If Windows uses a default context, the leak check races and may report leaks.  If Windows does not use a default context, it doesn't.  
But the call here to .release() vs. .reset() DOES determine whether the default context (if one exists) ACTUALLY leaks or not.  We want .reset() so that it is actually destroyed, otherwise it leaks.